javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -694,6 +694,10 @@ public boolean replace(Node node, Node replacementNode) {
      */
     public Node findRootNode() {
         Node n = this;
+        // (Non-orphan) comments are not integrated into the normal AST; we need to get the commented node first.
+        if (this instanceof Comment && ((Comment) this).getCommentedNode().isPresent()) {
+            n = ((Comment) this).getCommentedNode().get();
+        }
```

Comment:
This could be done with a method override in comment, or not?

Reply:
Uh oh, I completely agree!

I fixed that. Also, while doing so, I moved the special logic required when `setComment(Comment)` is called on a Comment node (-> comments on comments are not allowed) to the Comment class. There, we had exactly the same glitch: a `this instanceof Comment` was used rather than a proper method override. :-)